python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -74,6 +75,19 @@ def get_unique_id(self, *, parent_unique_id: Optional[str], index: Optional[int]
         parts = [str(parent_unique_id), str(index), self.__class__.__name__, self.description]
         return non_secure_md5_hash_str("".join(parts).encode())
 
+    def get_hash(
+        self, *, parent_unique_id: Optional[str] = None, index: Optional[int] = None
+    ) -> int:
+        """Generates a hash based off of the unique ids of all children."""
+        unique_id = self.get_unique_id(parent_unique_id=parent_unique_id, index=index)
+        hashes = [hash(unique_id)]
+        for i, child in enumerate(self.children):
+            hashes.append(child.get_hash(parent_unique_id=unique_id, index=i))
+        return hash(sum(hashes))
```

Comment:
I was curious about using `sum` for hash and found [this](https://docs.python.org/3/reference/datamodel.html#object.__hash__)

>  it is advised to mix together the hash values of the components of the object that also play a part in comparison of objects by packing them into a tuple and hashing the tuple

so probably worth changing this last line to `hash(tuple(hashses))`